data/transactions/verify/txn.go

```diff
@@ -54,23 +54,26 @@ type Context struct {
 // Group data are omitted because they are committed to in the
 // transaction and its ID.
 type Params struct {
-	CurrSpecAddrs transactions.SpecialAddresses
-	CurrProto     protocol.ConsensusVersion
+	CurrSpecAddrs  transactions.SpecialAddresses
+	CurrProto      protocol.ConsensusVersion
+	MinTealVersion uint64
 }
 
 // PrepareContexts prepares verification contexts for a transaction
 // group.
 func PrepareContexts(group []transactions.SignedTxn, contextHdr bookkeeping.BlockHeader) []Context {
 	ctxs := make([]Context, len(group))
+	minTealVersion := logic.ComputeMinTealVersion(group)
```

Comment:
this probably should be done lazely only if logic/app call txn in the group.
Or even done in LogicSigSanityCheck?

Reply:
I would prefer to leave this code very simple unless it slows things down too much. Transaction groups have at most 16 transactions, so I think this should be fast enough, but if it causes problems we can optimize.